locksmith/src/models/hook.ts

```diff
@@ -0,0 +1,37 @@
+import { Table, Column, Model, AllowNull, Validate } from 'sequelize-typescript'
+
+@Table({ tableName: 'Hooks', timestamps: true })
+// eslint-disable-next-line import/prefer-default-export
+export class Hook extends Model<Hook> {
+  @Column({ primaryKey: true, autoIncrement: true })
+  id!: number
+
+  @Column
+  network!: string
+
+  @Column
+  topic!: string
+
+  @AllowNull(true)
+  @Column
+  lock?: string
+
+  @Column
+  expiration!: Date
+
+  @Validate({
+    isIn: [['subscribe', 'unsubscribe']],
+  })
+  @Column
+  mode!: 'subscribe' | 'unsubscribe'
```

Comment:
I am not fully sure of the purpose here? Is it just to keep track of which ones have been subscribed/unsubscribed?

Reply:
Yes